app/Http/Controllers/HomeController.php

```diff
@@ -23,14 +24,20 @@ public function __construct()
      *
      * @return \Illuminate\Http\Response
      */
-    public function index()
+    public function index(Request $request)
     {
+        $count = isset($request['items_per_page']) ? $request['items_per_page'] : 5;
+        $articles = Article::whereNotNull('published_at')->orderBy('published_at', 'desc')->paginate($count);
```

Comment:
Please add model scope `scopePublished($query) {...}` for reusability and readability by calling `Article::published()`